src/Detector.php

```diff
@@ -67,19 +67,26 @@ public function determineMimeType(\SplFileObject $file): string
     /**
      * Read width and height from a image file.
      *
-     * @param string $filePath
+     * @param \SplFileObject $file File object
      * @return array [$width, $height]
      * @throws \InvalidArgumentException Unreadable SplFileInfo
      */
-    public function determineDimensions(string $filePath): array
+    public function determineDimensions(\SplFileObject $file): array
     {
-        $stream = new \SplFileInfo($filePath);
-        if (!$stream->isReadable()) {
-            throw new \InvalidArgumentException('File path ' . $filePath . ' is not existed');
+        if (false !== $path = $file->getRealPath()) {
+            $info = getimagesize($path);
```

Comment:
[`getimagesize`](https://secure.php.net/manual/en/function.getimagesize.php) may return `false` here (for whatever possible failure states exist). That case should probably be handled, otherwise `$info[0]` and `$info[1]` will throw.